python/google/protobuf/json_format.py

```diff
@@ -626,28 +626,28 @@ def _ConvertFieldValuePair(self, js, message, path):
                                          '{0}.{1}'.format(path, name)))
       except ParseError as e:
         if field and field.containing_oneof is None:
-          raise ParseError('Failed to parse {0} field: {1}.'.format(name, e))
+          raise ParseError('Failed to parse {0} field: {1}.'.format(name, e)) from e
         else:
-          raise ParseError(str(e))
+          raise ParseError(str(e)) from e
       except ValueError as e:
-        raise ParseError('Failed to parse {0} field: {1}.'.format(name, e))
+        raise ParseError('Failed to parse {0} field: {1}.'.format(name, e)) from e
       except TypeError as e:
-        raise ParseError('Failed to parse {0} field: {1}.'.format(name, e))
+        raise ParseError('Failed to parse {0} field: {1}.'.format(name, e)) from e
 
   def _ConvertAnyMessage(self, value, message, path):
     """Convert a JSON representation into Any message."""
     if isinstance(value, dict) and not value:
       return
     try:
       type_url = value['@type']
-    except KeyError:
+    except KeyError as e:
       raise ParseError(
-          '@type is missing when parsing any message at {0}'.format(path))
+          '@type is missing when parsing any message at {0}'.format(path)) from e
```

Comment:
This line is more than 80 columns wide; please shrink it. I suggest a line-break before path:
```
+          '@type is missing when parsing any message at {0}'.format(
+              path)) from e
```

Reply:
3087c139cb0d9812724a6417057853720ae4147e & 571893514cc27df5557de8fae93b66036205f671!